packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -112,4 +125,39 @@ export class EVMSmartWallet {
     public async nfts() {
         return this.crossmintService.fetchNFTs(this.address, this.chain);
     }
+
+    public async sendTransaction<
+        const TAbi extends Abi | readonly unknown[],
+        TFunctionName extends ContractFunctionName<TAbi, "nonpayable" | "payable"> = ContractFunctionName<
+            TAbi,
+            "nonpayable" | "payable"
+        >,
+        TArgs extends ContractFunctionArgs<TAbi, "nonpayable" | "payable", TFunctionName> = ContractFunctionArgs<
+            TAbi,
+            "nonpayable" | "payable",
+            TFunctionName
+        >
+    >({
+        address,
+        abi,
+        functionName,
+        args,
+        resendCallback,
+    }: Omit<
+        WriteContractParameters<TAbi, TFunctionName, TArgs, typeof this.accountClient.chain>,
+        "chain" | "account"
+    > & {
+        resendCallback?: TransactionServiceResendCallback;
+    }): Promise<Hex> {
```

Comment:
As may be seen in comments above, we were thinking that this function would simplify devx greatly if it can do simulation and confirmation all in one call

If we do this, though, we need a more complex return type, where we can return failure errors of each of the stages. Midn thinking about that?

Reply:
I think this function already does all of those. As for the return type, the idiom we're following in this code is to yield error information through typed exceptions. I think the two (very similar) options for the return type here are:
- Just the tx hash
- The transaction receipt

Reply:
Ok, that may work, but we should review the errors either here or in another PR. I would do it in this one (maybe in the TSDoc?)

Reply:
Current errors are in SendTransactionService.ts